pay/instructions.go

```diff
@@ -3,65 +3,22 @@ package pay
-const (
-	MethodKeyAny            cbc.Key = "any" // Use any method available.
-	MethodKeyCard           cbc.Key = "card"
-	MethodKeyCreditTransfer cbc.Key = "credit-transfer"
-	MethodKeyDebitTransfer  cbc.Key = "debit-transfer"
-	MethodKeyCash           cbc.Key = "cash"
-	MethodKeyCheque         cbc.Key = "cheque"
-	MethodKeyCredit         cbc.Key = "credit"
-	MethodKeyBankDraft      cbc.Key = "bank-draft"
-	MethodKeyDirectDebit    cbc.Key = "direct-debit" // aka. Mandate
-	MethodKeyOnline         cbc.Key = "online"       // Website from which payment can be made
-)
-
-// MethodKeyDef is used to define each of the Method Keys
-// that can be accepted by GOBL.
-type MethodKeyDef struct {
-	// Key being described
-	Key cbc.Key `json:"key" jsonschema:"title=Key"`
-	// Human value of the key
-	Title string `json:"title" jsonschema:"title=Title"`
-	// Details about the meaning of the key
-	Description string `json:"description" jsonschema:"title=Description"`
-	// UNTDID 4461 Equivalent Code
-	UNTDID4461 cbc.Code `json:"untdid4461" jsonschema:"title=UNTDID 4461 Code"`
-}
-
-// MethodKeyDefinitions includes all the payment method keys that
-// are accepted by GOBL.
-var MethodKeyDefinitions = []MethodKeyDef{
-	{MethodKeyAny, "Any", "Any method available, no preference.", "1"},                            // Instrument not defined
-	{MethodKeyCard, "Card", "Credit or debit card.", "48"},                                        // Bank card
-	{MethodKeyCreditTransfer, "Credit Transfer", "Sender initiated bank or wire transfer.", "30"}, // credit transfer
-	{MethodKeyDebitTransfer, "Debit Transfer", "Receiver initiated bank or wire transfer.", "31"}, // debit transfer
-	{MethodKeyCash, "Cash", "Cash in hand.", "10"},                                                // in cash
-	{MethodKeyCheque, "Cheque", "Cheque from bank.", ""},                                          // cheque
-	{MethodKeyCredit, "Credit", "Using credit from previous transactions with the supplier.", ""}, // credit
-	{MethodKeyBankDraft, "Draft", "Bankers Draft or Bank Cheque.", ""},                            // Banker's draft,
-	{MethodKeyDirectDebit, "Direct Debit", "Direct debit from the customers bank account.", "49"}, // direct debit
-	{MethodKeyOnline, "Online", "Online or web payment.", "68"},                                   // online payment service
-}
-
 // Instructions determine how the payment has or should be made. A
 // single "key" exists in which the preferred payment method
 // should be provided, all other details serve as a reference.
 type Instructions struct {
-	// How payment is expected or has been arranged to be collected
-	Key cbc.Key `json:"key" jsonschema:"title=Key"`
+	// The payment means expected or that have been arranged to be used to make the payment.
+	Key MeansKey `json:"key" jsonschema:"title=Key"`
+	// Additional code for the payment means if not defined in the standard list of keys.
+	Code cbc.Code `json:"code,omitempty" jsonschema:"title=Code"`
```

Comment:
If `MeansKey` is set to `other`, then I guess we need to ensure the `Code` is present, otherwise we wouldn't know how to map it to the FatturaPA `MP` code.  Should we provide a list of available `MP`'s?

Reply:
I see what you mean... the lack of a default in Italy could be confusing when set to `other`. So yes, we could add some validation there for Italy. I'll add that now. Nice catch!

Reply:
Okay, validation now included to make sure if `other` is selected in Italy, the `code` is also required.